test/common/src/main/kotlin/org/codice/compliance/verification/core/CoreVerifier.kt

```diff
@@ -20,12 +20,59 @@ import org.codice.compliance.SAMLCore_5_4_2_a
 import org.codice.compliance.SAMLCore_5_4_2_b
 import org.codice.compliance.SAMLCore_5_4_2_b1
 import org.codice.compliance.SAMLCore_6_1_b
+import org.codice.compliance.SAMLSpecRefMessage
 import org.codice.compliance.allChildren
 import org.codice.compliance.children
 import org.codice.compliance.utils.TestCommon.Companion.ELEMENT
+import org.codice.compliance.utils.TestCommon.Companion.REQUESTER
+import org.codice.compliance.utils.TestCommon.Companion.VERSION_MISMATCH
 import org.w3c.dom.Node
 
 class CoreVerifier(val node: Node) {
+
+    /**
+     * Verifies that a response has a status code of urn:oasis:names:tc:SAML:2.0:status:Requester.
+     * This should be called explicitly if an error of this type is expected.
+     */
+    fun verifyRequesterError(samlErrorCode: SAMLSpecRefMessage) {
+        verifyAndReturnStatus(samlErrorCode).children("Status Code").forEach {
+            val code = it.attributes.getNamedItem("Value").textContent
+            if (code != REQUESTER)
+                throw SAMLComplianceException.createWithPropertyMessage(samlErrorCode,
```

Comment:
✏️ I think it would be a good idea to put in some extra SAMLSpecRefMessages here. The ones that say things like
```
HTTP interactions during the message exchange MUST NOT use HTTP error status codes to indicate
failures in SAML processing, since the user agent is not a full party to the SAML protocol exchange.
```
So that if the IdP errors, the person running the test knows what part of the spec we are testing, along with the fact that they have to return a saml error response.

Reply:
So my one thing with that is, it's not part of the Core spec and by adding that we go away from the whole "separate verification by document" thing.

Reply:
Good point. Plus we SHOULD (<- now I'm sounding like the spec) be calling the http status code check beforehand anyway.

Reply:
Ooo, this is part of the core document and may be beneficial to put in the error codes:
```
If a SAML responder deems a request to be invalid according to SAML syntax or processing rules, then if it responds, it MUST return a SAML response message with a <StatusCode> element with the value urn:oasis:names:tc:SAML:2.0:status:Requester.
```

Reply:
True but now that it's generalized, you have to add that to the SAML Messages and pass it in if your test is expecting the `Requester` status code

Reply:
Or we can be sneaky...
`
If a SAML responder deems a request to be invalid according to SAML syntax or processing rules, then if it responds, it MUST return a SAML response message with a <StatusCode> element
`
or something of the sort.

Reply:
I just feel like we should them:
- What part of the spec we were testing when this happened
- And what they should do and why they should do it, according to the spec.

Reply:
@blen-desta and @bakejeyner what if, only in the Requester error message, we don't take a SamlSpecRefMessage, but instead just statically use the paragraph you referenced above ☝️ ?

Reply:
I care less about the version mismatch because that should only really affect like one test...

Reply:
I feel like it's a good idea to tell them where in the spec we were negative testing when that happened, as well as that above message. But that's just my opinion.

Reply:
That is what the code does. I prints the above message and whatever message you provide it (which will be "where in the spec we were negative testing")